portal/src/graphql/portal/EditOAuthClientForm.tsx

```diff
@@ -330,8 +348,13 @@ const EditOAuthClientForm: React.VFC<EditOAuthClientFormProps> =
 
     const showClientSecret = useMemo(
       () =>
-        clientConfig.x_application_type === "confidential" ||
-        clientConfig.x_application_type === "third_party_app",
+        (
```

Comment:
Custom UI Settings should not be shown for m2m client applications, as that feature is irrelevant to m2m.